content/en/dynamic_instrumentation/expression-language.md

```diff
@@ -0,0 +1,63 @@
+---
+title: Dynamic Instrumentation Expression Language
+kind: documentation
+private: false
+---
+
+The Dynamic Instrumentation Expression Language is a language that's designed to handle tasks such as formulating log probe message templates, metric probe expressions, span tag values, and probe conditions. It borrows syntax elements from common programming languages, but stands as its own entity with unique rules. The language allows for the accessing of local variables, method parameters, and nested fields within objects, and supports the use of comparison and logical operators. 
+
+For example, you can create a histogram from the size of a collection using `count(myCollection)` as the metric expression. Metric expressions must evaluate to a number.
+
+In log templates and tag values, expressions are delimited from the static parts of the template with brackets, for example: `User name is {user.name}`. Log template expressions can evaluate to any value. If evaluating the expression fails, it is replaced with `UNDEFINED`.
+
+Probe conditions must evaluate to a Boolean, for example: `startsWith(user.name, "abc")`, `len(str) > 20` or `a == b`.
+
+Some general guidelines on what the Expression Language supports:
+* You CAN access local variables, method parameters, and deeply nested fields/attributes within objects
+* You CAN use comparison operators (`<`, `>`, `>=`, `<=`, `==`, `!=`) to compare variables, fields, and constants in your conditions, as shown in the example `localVar1.field1.field2 != 15.`
+* You CAN use logical operators (`&&`, `||`, and `not` or `!`) to build complex Boolean conditions.
+* You CAN use the `null` literal (equivalent of `nil` in Python)
+* You CANNOT call methods, as Dynamic Instrumentation does not allow users to execute code that may have side effects.
```

Comment:
```suggestion
* You CANNOT call methods, as Dynamic Instrumentation does not allow users to execute code that may have side effects.
   * Alternatively: You CAN access `private` fields directly

```

Reply:
This suggestion was applied as part of 882852b920ceb87b41906dfaa54a92dce0c61743